README_CHUNK_FORMAT.rst

```diff
@@ -103,40 +92,108 @@ All entries are little endian.
+**Dictionary (optional)**
+
+*Only for C-Blosc2*
+
+Dictionaries are small datasets that are known to be repeated a lot and can help to compress data in blocks better.
+The dictionary section contains the size of the dictionary `int32_t dsize` followed by the dictionary data::
+
+    +=======+=================+
+    | dsize | dictionary data |
+    +=======+=================+
 
-    +========+=================+
-    | dsize  | dictionary data |
-    +========+=================+
+**Compressed Data Streams**
 
-Finally, it comes the actual list of compressed blocks / splits data streams.  It turns out that a block may optionally (see bit 4 in `flags` above) be further split in so-called splits which are the actual data streams that are transmitted to codecs for compression.  If a block is not split, then the split is equivalent to a whole block.  Before each split in the list, there is the compressed size of it, expressed as an `int32_t`::
+Each data stream contains the compressed size of the stream `int32 csize` followed by the data for the stream.
 
+If bit 4 of the `flags` header field is *not* set, each block of is stored in a single data stream::
+
+    +========+========+
+    | csize0 | cdata0 |
+    +========+========+
+    | block0          |
+    +========+========+
+
+If bit 4 of the `flags` header is set, each block can be stored using multiple data streams::
+
+    +========+========+========+========+========+========+========+
+    | csize0 | cdata0 | csize1 | cdata1 |   ...  | csizeN | cdataN |
     +========+========+========+========+========+========+========+
-    | csize0 | split0 | csize1 | split1 |   ...  | csizeN | splitN |
+    | block0                                                       |
     +========+========+========+========+========+========+========+
 
-[**Only for C-Blosc2**] The `csize` can be signed.  Positive values mean regular compressed sizes (the only ones supported by C-Blosc1).  Negative values mean splits that are made of a sequence of the same byte value; such value is encoded as the lowest significant byte of the `int32_t csize`.  For example, a csize of 10000 means that the compressed split stream that follows is 10000 bytes long.  On its hand, a csize of -32 means that the whole block/split is made of bytes with a value of 32.
+The uncompressed size for each block is equivalent to the `blocksize` field in the header, with the exception
+of the last block which may be equal to or less than the `blocksize`.
+
+**Run-Length Encoding**
+
+*Only for C-Blosc2*
+
+The `csize` field of each compressed data stream can be used to support run-length encoding for blocks as follows:
 
-*Note*: all the integers are stored in little endian.
\ No newline at end of file
+    - When the most significant bit is *not* set, `csize` represents the size of the compressed
+      data stream that follows. (supported by C-Blosc1)
```

Comment:
"supported by" -> "as in".  This is the only meaning for `csize` in C-Blosc1, so I think this reads better this way?